assets/src/blocks/amp-story-cta/edit.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import uuid from 'uuid/v4';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Component,
+	Fragment,
+} from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import {
+	Dashicon,
+	IconButton,
+	PanelBody,
+	withFallbackStyles,
+} from '@wordpress/components';
+import {
+	URLInput,
+	RichText,
+	ContrastChecker,
+	InspectorControls,
+	withColors,
+	PanelColorSettings,
+	withFontSizes,
+	FontSizePicker,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { FontFamilyPicker } from '../../components';
+import { maybeEnqueueFontStyle } from '../../helpers';
+
+const { getComputedStyle } = window;
+
+const applyFallbackStyles = withFallbackStyles( ( node, ownProps ) => {
+	const { textColor, backgroundColor } = ownProps.attributes;
+	const editableNode = node.querySelector( '[contenteditable="true"]' );
+	const computedStyles = editableNode ? getComputedStyle( editableNode ) : null;
+
+	return {
+		fallbackBackgroundColor: backgroundColor || ! computedStyles ? undefined : computedStyles.backgroundColor,
+		fallbackTextColor: textColor || ! computedStyles ? undefined : computedStyles.color,
+	};
+} );
+
+class CallToActionEdit extends Component {
+	constructor( props ) {
+		super( ...arguments );
+
+		if ( ! props.attributes.anchor ) {
+			this.props.setAttributes( { anchor: uuid() } );
+		}
+
+		this.nodeRef = null;
+		this.bindRef = this.bindRef.bind( this );
+	}
+
+	bindRef( node ) {
+		if ( ! node ) {
+			return;
+		}
+		this.nodeRef = node;
+	}
+
+	render() {
+		const {
+			attributes,
+			backgroundColor,
+			textColor,
+			setBackgroundColor,
+			setTextColor,
+			fallbackBackgroundColor,
+			fallbackTextColor,
+			setAttributes,
+			isSelected,
+			className,
+			fontSize,
+			setFontSize,
+		} = this.props;
+
+		const {
+			text,
+			url,
+			ampFontFamily,
+		} = attributes;
+
+		return (
+			<Fragment>
+				<div className={ className } ref={ this.bindRef }>
+					<InspectorControls>
+						<PanelBody title={ __( 'Text Settings', 'amp' ) }>
+							<FontFamilyPicker
+								name={ ampFontFamily }
+								onChange={ ( value ) => {
+									maybeEnqueueFontStyle( value );
+									setAttributes( { ampFontFamily: value } );
+								} }
+							/>
+							<FontSizePicker
+								value={ fontSize.size }
+								onChange={ setFontSize }
+							/>
+						</PanelBody>
+						<PanelColorSettings
```

Comment:
Probably the repeating settings with Text Block we could at some point move to a separate file, maybe this could also have background opacity setting as well. Later.

Reply:
Good idea. Once both PRs are merged, we can move this into a separate `BlockColorSettings` component or something.